pytest_pyvista.py

```diff
@@ -1,21 +1,168 @@
 # -*- coding: utf-8 -*-
 
+import os
+import pathlib
+import platform
+import warnings
+
 import pytest
+import pyvista
+from pyvista._vtk import VTK9
 
 
 def pytest_addoption(parser):
-    group = parser.getgroup('pyvista')
+    """Adds new flag options to the pyvista plugin."""
+
+    group = parser.getgroup("pyvista")
+    group.addoption(
+        "--reset_image_cache",
+        action="store_true",
+        help="Reset the images in the PyVista cache.",
+    )
     group.addoption(
-        '--foo',
-        action='store',
-        dest='dest_foo',
-        default='2022',
-        help='Set the value for the fixture "bar".'
+        "--ignore_image_cache", action="store_true", help="Ignores the image cache."
     )
+    group.addoption(
+        "--fail_extra_image_cache",
+        action="store_true",
+        help="Enables failure if image cache does not exist.",
+    )
+    parser.addini("HELLO", "Dummy pytest.ini setting")
+
+
+class VerifyImageCache:
+    """Control image caching for testing.
+
+    Image cache files are names according to ``test_name``.
+    Multiple calls to an instance of this class will append
+    `_X` to the name after the first one.  That is, files
+    ``{test_name}``, ``{test_name}_1``, and ``{test_name}_2``
+    will be saved if called 3 times.
+
+    Parameters
+    ----------
+    test_name : str
+        Name of test to save.  Sets name of image cache file.
+
```

Comment:
Some parameters such as `cache_dir`, `error_value` are missing in the documentation.

Reply:
Done. 

`cache_dir` was moved to the pytest command line, which I think will be more useful.